File: src/json2graph/domain/create_new_migration_parameters.py
from pydantic import BaseModel

from json2graph.domain.migration_type_enum import MigrationType


class CreateNewMigrationParameters(BaseModel, use_enum_values=True):
    migration_name: str
    mongodb_url: str
    mongodb_database: str
    mongodb_collection: str
    migration_type: MigrationType
    root_database_name: str | None = None
    mongodb_producer_replicas: int = 0
    json_to_node_replicas: int = 0
    node_to_neo4j_replicas: int = 0
    neo4j_create_relationship_replicas: int = 0
    mongodb_producer_limit: int | None = None

File: frontend/data2graph/src/domain/create-new-migration-parameters.ts
import { MigrationType } from "./migration-type";

export type CreateNewMigrationParameters = {
    migration_name: string;
    mongodb_url: string;
    mongodb_database: string;
    mongodb_collection: string;
    migration_type: MigrationType;
    root_database_name?: string;
    mongodb_producer_replicas?: number;
    json_to_node_replicas?: number;
    node_to_neo4j_replicas?: number;
    neo4j_create_relationship_replicas?: number;
}
